src/plugins/expressions/public/react_expression_renderer.tsx

```diff
@@ -30,7 +30,11 @@ export interface ReactExpressionRendererProps extends IExpressionLoaderParams {
   ) => React.ReactElement | React.ReactElement[];
   padding?: 'xs' | 's' | 'm' | 'l' | 'xl';
   onEvent?: (event: ExpressionRendererEvent) => void;
-  onData$?: <TData, TInspectorAdapters>(data: TData, adapters?: TInspectorAdapters) => void;
+  onData$?: <TData, TInspectorAdapters>(
```

Comment:
we need a prop to allow opting in to partial result. we should pass that to loader then, if prop is not enabled partial data should not be rendered (only final results)